godal.cpp

```diff
@@ -1766,5 +1766,71 @@ GDALDatasetH godalNearblack(cctx *ctx, const char *pszDest, GDALDatasetH hDstDS,
 	}
 
 	godalUnwrap();
+	return ret;
+}
+
+OGRSpatialReferenceH godalGetGCPSpatialRef(GDALDatasetH hSrcDS) {
+	return GDALGetGCPSpatialRef(hSrcDS);
+}
+
+int godalGetGCPCount(GDALDatasetH hSrcDS) {
+	return GDALGetGCPCount(hSrcDS);
+}
+
+const GDAL_GCP *godalGetGCPs(GDALDatasetH hSrcDS) {
+	return GDALGetGCPs(hSrcDS);
+}
+
+const char *godalGetGCPProjection(GDALDatasetH hSrcDS) {
+	return GDALGetGCPProjection(hSrcDS);
+}
+
+void godalSetGCPs(cctx *ctx, GDALDatasetH hSrcDS, int numGCPs, GDAL_GCP *GCPList, const char *pszGCPProjection) {
+	godalWrap(ctx);
+
+	CPLErr ret = GDALSetGCPs(hSrcDS, numGCPs, GCPList, pszGCPProjection);
+	if(ret!=0) {
+		forceCPLError(ctx, ret);
+	}
+
+	GDALDeinitGCPs(numGCPs, GCPList);
+	CPLFree(GCPList);
+	
+	godalUnwrap();
+	return;
+}
+
+void godalSetGCPs2(cctx *ctx, GDALDatasetH hSrcDS, int numGCPs, GDAL_GCP *GCPList, OGRSpatialReferenceH hSRS) {
+	godalWrap(ctx);
+
+	CPLErr ret = GDALSetGCPs2(hSrcDS, numGCPs, GCPList, hSRS);
+	if(ret!=0) {
+		forceCPLError(ctx, ret);
+	}
+
+	GDALDeinitGCPs(numGCPs, GCPList);
+	CPLFree(GCPList);
+	
+	godalUnwrap();
+	return;
+}
+
+// Utility function to aid conversion from Go `GCP` type to `GDAL_GCP*`, allocates memory for `GDAL_GCP*` and requires a manual `CPLFree`
```

Comment:
Ideally, this function should be removed and merged into `godalSetGCPs`, with `godalSetGCPs` signature changed to accept our own C.goGCPList struct instead of a `*GDAL_GCP`. This would also allow calling `GDALInitGCPs` and `GDALDeInitGCPs` in the same function which is cleaner / easier to understand